src/Razor/src/Microsoft.CodeAnalysis.Razor.CohostingShared/RazorAnalyzerAssemblyRedirector.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Composition;
+using IRazorAnalyzerAssemblyRedirector = Microsoft.CodeAnalysis.ExternalAccess.Razor.RazorAnalyzerAssemblyRedirector.IRazorAnalyzerAssemblyRedirector;
+
+namespace Microsoft.VisualStudio.Razor;
+
+#pragma warning disable RS0030 // Do not use banned APIs
+[Shared]
+[Export(typeof(IRazorAnalyzerAssemblyRedirector))]
+[method: ImportingConstructor]
+#pragma warning restore RS0030 // Do not use banned APIs
+internal class RazorAnalyzerAssemblyRedirector() : IRazorAnalyzerAssemblyRedirector
+{
+    private static readonly ImmutableArray<Type> s_compilerAssemblyTypes = [
+
+        typeof(CodeAnalysis.Razor.CompilerFeatures), // Microsoft.CodeAnalysis.Razor.Compiler.dll
+        typeof(AspNetCore.Razor.ArgHelper), // Microsoft.AspNetCore.Razor.Utilities.Shared.dll
+
+        // The following dependencies will be provided by the Compiler ALC so its not strictly required to redirect them, but we do so for completeness. 
+        typeof(Microsoft.Extensions.ObjectPool.ObjectPool), // Microsoft.Extensions.ObjectPool.dll
+        typeof(ImmutableArray) // System.Collections.Immutable.dll
+    ];
+
+    private static readonly ImmutableDictionary<string, string> s_compilerAssemblyMap = s_compilerAssemblyTypes.ToImmutableDictionary(t => t.Assembly.GetName().Name!, t => GetAssemblyLocation(t.Assembly));
```

Comment:
Please consider a `FrozenDictionary<string, string>` rather than paying for `ImmutableDictionary<string, string>` lookup.